### PR TITLE
8 get/api/articles

### PR DESCRIPTION
**Pull request by h-chagas**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -34,111 +34,167 @@ describe("GET /api/topics", () => {
...
+   test("200: responds with an array of articles sorted by date in descending order", () => {
+      return request(app)
+         .get("/api/articles")
+         .expect(200)
+         .then((response) => {
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
+            expect(response.body).toBeSortedBy('created_at', {descending: true,})
+            console.log(response.body, '<<< my response.body')
```
Comment (leekli) [inline __tests__/app.test.js]: Please remove this console log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -34,111 +34,167 @@ describe("GET /api/topics", () => {
...
+   });
+   test("200: filter objects by topic given", () => {
+    return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
         .then((response) => {
-            expect(response.body.msg).toBe('Please type/select a number to choose an article')
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
```
Comment (leekli) [inline __tests__/app.test.js]: I would make this expect more specific, for example how many articles should come back with the topic of cats? This expect statement doesn't tell me anything other than it's greater than 0, however for a more confident test let's see a number here which is representative of how many actual articles are returned... e.g. if there are 3 articles with the topic of 'cats' in the database, then this expect statement should be `expect(response.body.length).toBe(3)`; or you could use another jest matcher `expect(response.body).toHaveLength(3)`

File: controllers/topics.controllers.js [hunk 3]
```diff
@@ -2,8 +2,11 @@ const { showTopics } = require('../models/topics.models')
 
 
 exports.getTopics = (req, res, next) => {
-    showTopics().then( (topics) => {
+    showTopics()
+    .then( (topics) => {
         res.status(200).send( topics )
```
Comment (leekli) [inline controllers/topics.controllers.js:7]: Does this work? Shouldn't the .send() have an object inside it? such as `{ topics: topics }`

File: __tests__/app.test.js [hunk 4]
```diff
@@ -34,111 +34,167 @@ describe("GET /api/topics", () => {
...
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    comment_count: expect.any(Number),
+                 });
+            })
+                        
         })
-    })
+   })
+
```
Comment (leekli) [inline __tests__/app.test.js:220]: I'd like to see a test for a topic it cannot find (a topic which does not exist in the database) - 404 error

Also how about a test for an invalid topic, code 400.

Also how about a topic which exists, but has no articles attached to it, what would we want that test to look like

**Review by leekli — CHANGES_REQUESTED** [review]
Well done so far, I have some Request for changes:

See the comments
* I'd like to see some additional tests
* See line 183 for my comment on the expect statement using .length
* Remove a console log

Once you have completed these actions, then please submit a new pull request review.

File: __tests__/app.test.js (hunk 1 above)
Comment (h-chagas) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 2 above)
Comment (h-chagas) [reply __tests__/app.test.js]: It makes total sense. I will study more about jest matchers this weekend!
Done

File: controllers/topics.controllers.js (hunk 3 above)
Comment (h-chagas) [reply controllers/topics.controllers.js:7]: It's working, as it is sending an array.

File: __tests__/app.test.js (hunk 4 above)
Comment (h-chagas) [reply __tests__/app.test.js:220]: done

**Review by leekli — APPROVED** [review]
Thanks for making these changes, Hugo - well done!

Please feel free to now merge this ticket/branch.

**Pull request merged**